### ISSUE TITLE
Unable To Unlock Settings In RustDesk App

### ISSUE THREAD

**Issue by Majoraslayer** [JReming85/rustdesk-kasm-image#2]
In the Security and Network settings of the RustDesk app, there's an Unlock button you have to hit to be able to change anything. I believe normally when I try to run this on a regular desktop it will pop up a password prompt for your Linux account, then after you enter it the settings will be unlocked. Clicking the Unlock button inside this container doesn't seem to be doing anything. I opened Terminal inside the container's desktop and made sure my account had a password set, logged out, deleted the session, and created a new session. It's still not giving me the password prompt to be able to unlock the settings.

**Comment by JReming85** [comment]
Good catch, not sure why I never saw this or maybe my settings just carried over to the new version.

Until I have time to look into it or find a work around you can try using the older version.

Either use one of the first github images or check out my gitlab
https://gitlab.com/jeremyrem/rustdesk-kasm-image/container_registry/4372146

I am fairly confident I left main as the old version and latest and the newest

**Comment by Majoraslayer** [comment]
> Good catch, not sure why I never saw this or maybe my settings just carried over to the new version.
> 
> Until I have time to look into it or find a work around you can try using the older version.
> 
> Either use one of the first github images or check out my gitlab https://gitlab.com/jeremyrem/rustdesk-kasm-image/container_registry/4372146
> 
> I am fairly confident I left main as the old version and latest and the newest

Unfortunately I was already running Main.  Any chance you do have a backup of an old version you could add under a new tag, such as Legacy?

**Comment by JReming85** [comment]
The main on github is the newer one, on gitlab I think its the old version.

Give me a moment to try and export it from my kasm server, since I still run it myself

**Comment by JReming85** [comment]
Found it and exported it using portainer, not really sure how to go forward from here without portainer but here is the tar file.
https://1drv.ms/f/s!AkVWMpsVqI5mi6IEJTil4gmcvzRMCA?e=Hu1WU4

You can also check one of these to see if it helps 
https://github.com/JReming85/rustdesk-kasm-image/pkgs/container/rustdesk-kasm-image/versions?page=5

or run one of the docker ubuntu images. Those give full root, and install rustdesk to configure it and set the persistant folder to the same as rustdesk so when you start Rustdesk it will read the edited config.

**Comment by JReming85** [comment]
Your in luck, just pushed it from portainer
https://github.com/JReming85/rustdesk-kasm-image/pkgs/container/rustdesk-kasm-image/185843675?tag=legacy

**Comment by Majoraslayer** [comment]
> Your in luck, just pushed it from portainer https://github.com/JReming85/rustdesk-kasm-image/pkgs/container/rustdesk-kasm-image/185843675?tag=legacy

Nice, thanks a ton!

**Comment by saukkico** [comment]
I wasn't able to launch the legacy client (I opened a new issue #3 ), so I had to figure out something else. This is how I was able to configure the network settings in the Rustdesk Latest Client:

You need to have the Rustdesk server config string. You can get it from a working Rustdesk client by clicking the [Export Server Config](https://rustdesk.com/docs/en/self-host/client-configuration/#3-setup-using-import-or-export) button on the network settings page. 

First I installed sudo to the container and configured it to reguire no password with instructions found [here](https://kasmweb.com/docs/latest/how_to/running_as_root.html). So basically I copied this to the "Docker exec config"- field in the Kasm Workspace settings

`{
 "first_launch":{
      "user":"root",
      "cmd":"bash -c 'apt-get update && apt-get install -y sudo && echo \"kasm-user  ALL=(ALL) NOPASSWD: ALL\" >> /etc/sudoers'"
  }
}`

Then I started the Container and opened Terminal and entered this command with the server config string. Remove all "=" characters from the beginning of the string.

`sudo /usr/lib/rustdesk/rustdesk --config server-config-string_without_the_=`

Then close the Rustdesk window and start again from the shortcut on the desktop and check if there are some settings in the network settings.

This worked for me and I was able to connect to my private Rustdesk server.
